sympy/physics/mechanics/body.py

```diff
@@ -562,3 +562,368 @@ def dcm(self, body):
         elif isinstance(body, Body):
             frame = body.frame
         return self.frame.dcm(frame)
+
+    def orient_axis(self, parent, axis, angle):
+        """Sets the orientation of this body with respect to a parent body or
+        reference frame by rotating through an angle about an axis fixed
+        in the parent reference frame or body.
+
+        Parameters
+        ==========
+
+        parent : Body or ReferenceFrame
+            Body or Reference frame that this body will be rotated relative
+            to.
+        axis : Vector
+            Vector fixed in the parent frame about which this body is
+            rotated. It need not be a unit vector and the rotation follows the
+            right hand rule.
+        angle : sympifiable
+            Angle in radians by which the frame is to be rotated.
+
+        Warns
+        ======
+
+        UserWarning
+            If the orientation creates a kinematic loop.
+
+        Examples
+        ========
+
+        Setup variables for the examples:
+
+        >>> from sympy import symbols
+        >>> from sympy.physics.mechanics import Body
+        >>> q1 = symbols('q1')
+        >>> A = Body('A')
+        >>> B = Body('B')
+        >>> B.orient_axis(A, A.x, q1)
+
+        The ``orient_axis()`` method generates a direction cosine matrix and
+        its transpose which defines the orientation of B relative to A and vice
+        versa. Once orient is called, ``dcm()`` outputs the appropriate
+        direction cosine matrix:
+
+        >>> B.dcm(A)
+        Matrix([
+        [1,       0,      0],
+        [0,  cos(q1), sin(q1)],
+        [0, -sin(q1), cos(q1)]])
+        >>> A.dcm(B)
+        Matrix([
+        [1,       0,        0],
+        [0, cos(q1), -sin(q1)],
+        [0, sin(q1),  cos(q1)]])
+
+        The following two lines show that the sense of the rotation can be
+        defined by negating the vector direction or the angle. Both lines
+        produce the same result.
+
+        >>> B.orient_axis(A, -A.x, q1)
+        >>> B.orient_axis(A, A.x, -q1)
+
+        """
+
+        if isinstance(parent, Body):
+            frame = parent.frame
+        elif isinstance(parent, ReferenceFrame):
+            frame = parent
+        self.frame.orient_axis(frame, axis, angle)
```

Comment:
Regarding https://github.com/sympy/sympy/pull/21988#issuecomment-909147200

This code can be
```suggestion
        if isinstance(parent, Body):
            parent = parent.frame
        self.frame.orient_axis(parent, axis, angle)
```

This way the `Body` is turned into frame in parent variable and `orient` function would handle proper error mg.

Reply:
Correct it everywhere.

Reply:
Done.